### PR TITLE
refactor(sozo): manifest and world address logic

### PR DESCRIPTION
**Pull request by lambda-0x**
Fix: #794 

it will try to derive path (in specified order) for

`manifest.json` from :

- passed cli argument
- Environment variable (`DOJO_MANIFEST_FILE`)
- `Scarb.toml`'s target directory and profile information

`world_address` from:

- passed cli argument
- Environment variable (`DOJO_WORLD_ADDRESS`)
- `Scarb.toml`'s dojo.tool.env
- manifest.json

### CODE REVIEW

File: crates/torii/server/src/cli.rs [hunk 1]
```diff
@@ -154,3 +121,73 @@ async fn main() -> anyhow::Result<()> {
...
+        Manifest::load_from_path(manifest_path)?
+    } else if ws.is_some() {
+        let ws = ws.as_ref().unwrap();
+        let target_dir = ws.target_dir().path_existent()?;
+        let target_dir = target_dir.join(ws.config().profile().as_str());
+        let manifest_path = target_dir.join("manifest.json");
+        Manifest::load_from_path(manifest_path)?
+    } else {
+        Manifest::load_from_path(Utf8PathBuf::from("./target/dev/manifest.json"))?
+    };
```
Comment (kariy) [inline crates/torii/server/src/cli.rs:158]: Whats the reasoning for defaulting to `target/dev/manifest.json` here?

To assume that `target` dir exists would also to assume that we're already in a `scarb` workspace. Don't think there'd be any situation where `target/dev/` would exists without a `scarb` workspace in the context of Dojo.

 Would be redundant when we already checking if `scarb` exists.

File: crates/torii/server/src/cli.rs [hunk 2]
```diff
@@ -154,3 +121,73 @@ async fn main() -> anyhow::Result<()> {
...
+// else uses scarb manifest to derive path of dojo manifest file,
+// else try to derive manifest path from scarb manifest
+// else try `./target/dev/manifest.json` as dojo manifest path
+//
+// If neither of this work return an error and exit
+fn get_manifest_and_env(
+    args_path: Option<&Utf8PathBuf>,
+) -> anyhow::Result<(Manifest, Option<Environment>)> {
+    // If there is scarb manifest file it shoul
+    let config;
```
Comment (kariy) [inline crates/torii/server/src/cli.rs:137]: this var doesn't seem to be used anywhere else apart from the `if` branch below.

File: crates/torii/server/src/cli.rs [hunk 3]
```diff
@@ -154,3 +121,73 @@ async fn main() -> anyhow::Result<()> {
...
+// Use manifest path from cli args,
+// else uses scarb manifest to derive path of dojo manifest file,
+// else try to derive manifest path from scarb manifest
+// else try `./target/dev/manifest.json` as dojo manifest path
+//
+// If neither of this work return an error and exit
+fn get_manifest_and_env(
+    args_path: Option<&Utf8PathBuf>,
+) -> anyhow::Result<(Manifest, Option<Environment>)> {
+    // If there is scarb manifest file it shoul
```
[the remark this answers is not in the document]
Comment (lambda-0x) [reply crates/torii/server/src/cli.rs:136]: forgot to remove it, will do

File: crates/torii/server/src/cli.rs (hunk 1 above)
Comment (lambda-0x) [reply crates/torii/server/src/cli.rs:158]: make sense, will remove

**Comment by lambda-0x** [comment]
(also added environment variables as per @tarrencev's suggestions)

**Comment by broody** [comment]
Works like a charm! ty ser

**Review by broody — APPROVED** [review]

**Pull request merged**